### DISCUSSION TITLE
How to provide a human-readable name of an inline Union type?

### DISCUSSION THREAD

**Discussion by sascha-wolf** [microsoft/typespec#3147] [Q&A]
To make a long story short: we've a custom `ID` scalar that we use to denote that a given property is a unique ID for a given model:

```
@friendlyName("{name}.id", M)
@doc("Unique ID for a {name}", M)
scalar ID<M extends {}> extends string;
```

This works great for cases like the following but breaks down for unions:
```
// (Thing.id)
// Unique ID for a Thing
@path thingID: ID<Thing>

// (undefined.id)
// Unique ID for a undefined
@path parentID: ID<Thing | AnotherThing>
```

I've scoured the documentation, even the TypeScript type definitions but to no avail. Am I right to assume that this is currently not supported?

**Comment by timotheeguerin** [comment]
Yeah it is not possible today, `{name}` interpolate the internal `name` of the `M` type which anonymous union don't have.Which emitter are you using? openapi3?
We do have some issue to provide a more standard name generation for templates, for example the json schema emitter does a better job at finding a name automatically.

**Reply by sascha-wolf** [reply]
Yes, we're using the OpenAPI3 emitter.

**Reply by sascha-wolf** [reply]
For the time being it'd even be helpful to be able to include the verbatim definition in there. Yes, `Thing | AnotherThing.id` isn't quite correct but `Unique ID for a Thing | AnotherThing` would already be helpful.
